src/base.c

```diff
@@ -25,15 +25,17 @@ base_map(extent_hooks_t *extent_hooks, unsigned ind, size_t size) {
 	if (extent_hooks == &extent_hooks_default) {
 		addr = extent_alloc_mmap(NULL, size, PAGE, &zero, &commit);
 	} else {
+		pre_reentrancy(tsdn_tsd(tsdn));
```

Comment:
The tsdn_tsd makes me a little nervous here - how do we known tsd has been initialized here? And if we do, then why not have the base code take a raw tsd instead?